### PR TITLE
Port robot tests

### PR DESCRIPTION
**Pull request by 1letter**
This is part of https://github.com/plone/Products.CMFPlone/issues/3813

port robot tests

- [x] [test_actionmenu.robot](https://github.com/plone/Products.CMFPlone/commit/e0718d9d659b866c4402035b869e562ba4980609)
- [x] [test_contentbrowser.robot](https://github.com/plone/Products.CMFPlone/commit/24c042bfc0a93209ec8667f2a5da819fed1a6c61)
- [x] [test_controlpanel_actionmenu.robot](https://github.com/plone/Products.CMFPlone/commit/522f942b5e4c65d531cbd76f3dc9536aca18303f)
- [x] [test_controlpanel_editing.robot](https://github.com/plone/Products.CMFPlone/commit/936cc36d6be4cc49500452b3efc1907a273e5d1f)
- [x] [test_controlpanel_filter.robot](https://github.com/plone/Products.CMFPlone/commit/8a26de1d04f05dbdf3427c21312c541546919519)
- [x] [test_controlpanel_language.robot](https://github.com/plone/Products.CMFPlone/commit/12746f26ed1cd16c5a64574e4b7ce3e4e36ec3ee)
- [x] [test_controlpanel_markup.robot](https://github.com/plone/Products.CMFPlone/commit/53f5956882dcda9be412f61d78332649deb78d48)
- [x] [test_controlpanel_navigation.robot](https://github.com/plone/Products.CMFPlone/commit/f0a70d1aa636fa448c14a799a1a26493a7c99aa8)
- [x] [test_controlpanel_redirection.robot](https://github.com/plone/Products.CMFPlone/commit/73677a5bf5e98e20afcac19389ae01479ecd6326)
- [x] [test_controlpanel_search.robot](https://github.com/plone/Products.CMFPlone/commit/9826bf0b73e9b022ff029c6f5c2714473ae5fecc)
- [x] [test_controlpanel_security.robot](https://github.com/plone/Products.CMFPlone/commit/caba91ed383980a42e186c626c89f42edb8656d9)
- [x] [test_controlpanel_site.robot](https://github.com/plone/Products.CMFPlone/commit/a6b2e810145aaee377dcdafb853db35bee4a9adf)
- [x] [test_controlpanel_social.robot](https://github.com/plone/Products.CMFPlone/commit/ceaacf758498eab8361141850bf78cef8bd290c4)
- [x] [test_controlpanel_types.robot](https://github.com/plone/Products.CMFPlone/commit/76112f5a2dc4d150fdf6e06e04d88bef17606ad8)
- [x] [test_controlpanel_usergroups.robot](https://github.com/plone/Products.CMFPlone/commit/bb3b474cde92d29c9b8b5f23b4a760cb6281fb84)
- [x] [test_edit.robot](https://github.com/plone/Products.CMFPlone/commit/80134a3a296eff60fc6cb6f82f96ef533b3ea7b9)
- [x] [test_edit_user_schema.robot](https://github.com/plone/Products.CMFPlone/commit/85903812aea4cf29a3088d476279a684f35a3ab4)
- [x] [test_folder_contents.robot](https://github.com/plone/Products.CMFPlone/commit/b61e30275dff7cbe86d2c3e0fb85f83cac2021ff)
- [x] [test_linkintegrity.robot](https://github.com/plone/Products.CMFPlone/commit/07de7e80aa7acc9d163458ce3168efa37ae09691)
- [x] [test_livesearch.robot](https://github.com/plone/Products.CMFPlone/commit/14ed1a09bea8d099baa7568dfde72f3292f58c83)
- [x] [test_overlays.robot](https://github.com/plone/Products.CMFPlone/commit/562d98888ecea785f9e3e4410efa37dd63d95835)
- [x] [test_portlets.robot](https://github.com/plone/Products.CMFPlone/pull/4019/commits/a7c3b0350f8bf94454634dc367f90871eb0e7baa)
- [x] [test_querystring.robot](https://github.com/plone/Products.CMFPlone/pull/4019/commits/1e9e137e77f58dee84589a87a1b2b37be7de7d10)
- [x] [test_tinymce.robot](https://github.com/plone/Products.CMFPlone/pull/4019/commits/3828732a6e8ec95d06ad0656728a525c487ce7a2)
- [x] variables.py
- [x] keywords.robot - should be marked as ready at the end
- [x] common.robot - possible obsolete, only in robodoc used. is this really needed?

### CODE REVIEW

**Comment by petschki** [comment]
@gforcada we're almost through with porting CMFPlone robot tests to playwright but right now the `rfbrowser init` fails on jenkins. We use this with the coredev PR here: https://github.com/plone/buildout.coredev/pull/954 to get the latest versions for `robotframework-browser` .. the error on the jenkins console says:

```
2024-10-07 05:27:33,807 [INFO    ] Traceback (most recent call last):
  File "/home/jenkins/.buildout/eggs/cp312/robotframework_browser-17.5.2-py3.12.egg/Browser/entry.py", line 324, in init
    _rfbrowser_init(skip_browsers, silent_mode, with_deps, browser)
  File "/home/jenkins/.buildout/eggs/cp312/robotframework_browser-17.5.2-py3.12.egg/Browser/entry.py", line 131, in _rfbrowser_init
    _check_npm()
  File "/home/jenkins/.buildout/eggs/cp312/robotframework_browser-17.5.2-py3.12.egg/Browser/entry.py", line 123, in _check_npm
    raise exception
  File "/home/jenkins/.buildout/eggs/cp312/robotframework_browser-17.5.2-py3.12.egg/Browser/entry.py", line 111, in _check_npm
    subprocess.run(
  File "/srv/python3.12/lib/python3.12/subprocess.py", line 548, in run
    with Popen(*popenargs, **kwargs) as process:
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/srv/python3.12/lib/python3.12/subprocess.py", line 1026, in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,
  File "/srv/python3.12/lib/python3.12/subprocess.py", line 1955, in _execute_child
    raise child_exception_type(errno_num, err_msg, err_filename)
FileNotFoundError: [Errno 2] No such file or directory: 'npm'
```

do you have an idea how to fix this?

**Comment by mauritsvanrees** [comment]
I think they can be removed yes.

@stevepiercy Is the [`robodoc` directory](https://github.com/plone/Products.CMFPlone/tree/840f075613c764036b773a4a93b11b036623130b/Products/CMFPlone/tests/robot/robodoc) used anywhere in current docs?  I don't think so.  Apparently it was originally some doctests used for generating screen shots.  See its [readme](https://github.com/plone/Products.CMFPlone/blob/840f075613c764036b773a4a93b11b036623130b/Products/CMFPlone/tests/robot/robodoc/README.rst).

Looking at the [history](https://github.com/plone/Products.CMFPlone/commits/840f075613c764036b773a4a93b11b036623130b/Products/CMFPlone/tests/robot/robodoc), the last real commit was from [2018](https://github.com/plone/Products.CMFPlone/commit/43f7bb0b8a1fecc09f2e9053cc0df87c6b11837e), all commits afterwards look more like search-and-replace through the entire code base.

**Comment by stevepiercy** [comment]
> I think they can be removed yes.
> 
> @stevepiercy Is the [`robodoc` directory](https://github.com/plone/Products.CMFPlone/tree/840f075613c764036b773a4a93b11b036623130b/Products/CMFPlone/tests/robot/robodoc) used anywhere in current docs? I don't think so. Apparently it was originally some doctests used for generating screen shots. See its [readme](https://github.com/plone/Products.CMFPlone/blob/840f075613c764036b773a4a93b11b036623130b/Products/CMFPlone/tests/robot/robodoc/README.rst).

There is one mention in the docs regarding usage. Does this need to be updated?

- https://6.docs.plone.org/contributing/core/index.html#test-locally

There is no documentation about how to use these changes in whatever context they are intended to be used. Not even a README. What's up with that?

**Comment by datakurre** [comment]
> @petschki @mauritsvanrees @datakurre
> robodoc ist obsolete, right? Should the entire folder be removed`/Products/CMFPlone/tests/robot/robodoc` ?

I second to remove.

**Comment by wesleybl** [comment]
> the error on the jenkins console says:

It looks like `node` is not installed.

**Comment by wesleybl** [comment]
> the error on the jenkins console says:

It seems that `node` is not installed only on Node1. On Node3 and Node4 it is installed.

**Comment by petschki** [comment]
## I have now tested this on all 3 nodes:

Node 1: https://jenkins.plone.org/job/pull-request-6.1-3.12/726/
Node 3: https://jenkins.plone.org/job/pull-request-6.1-3.12/728/
Node 4: https://jenkins.plone.org/job/pull-request-6.1-3.12/727/

### Result: 

Node 1:

- no node available

```
FileNotFoundError: [Errno 2] No such file or directory: 'npm' 
```

Node 3 and 4:

- node outdated

```
...
2024-10-10 22:24:34,966 [INFO    ] npm WARN EBADENGINE Unsupported engine {
2024-10-10 22:24:34,966 [INFO    ] npm WARN EBADENGINE   package: '@eslint/config-array@0.18.0',
2024-10-10 22:24:34,966 [INFO    ] npm WARN EBADENGINE   required: { node: '^18.18.0 || ^20.9.0 || >=21.1.0' },
2024-10-10 22:24:34,966 [INFO    ] npm WARN EBADENGINE   current: { node: 'v20.5.1', npm: '9.8.0' }
2024-10-10 22:24:34,966 [INFO    ] npm WARN EBADENGINE }
...
```

@gforcada FYI

**Comment by petschki** [comment]
Node 3: https://jenkins.plone.org/job/pull-request-6.1-3.10/441/console
Node 4: https://jenkins.plone.org/job/pull-request-6.1-3.12/735/console

Node 1 still has problems starting `rfbrowser init`

**Comment by petschki** [comment]
@gforcada 

I tried to run this PR in combination with the version updates from coredev but there's the error:
```
+ git checkout robotframework-browser-versionupdates
error: Your local changes to the following files would be overwritten by checkout:
	checkouts.cfg
```

does jenkins some changes to `checkouts.cfg` during installation ?

@1letter there are robottest errors which I didn't see locally. Do you have an idea what could be wrong there?
See https://jenkins.plone.org/job/pull-request-6.1-3.12/736/console

**Comment by 1letter** [comment]
Looks like a old version of robotframework browser library, for example the `test_controlpanel_site.robot` use this `Download    ${PLONE_URL}/sitemap.xml.gz    saveAs=/tmp/sitemap.xml.gz`, but in version robotframework browser library < 18.0 is only one parameter allowed.

**Comment by gforcada** [comment]
@petschki yes, I think we add/remove things there... but your test was about which branches? 🤔

**Comment by petschki** [comment]
> @petschki yes, I think we add/remove things there... but your test was about which branches? 🤔

those PRs were tested (see job parameters):

```
https://github.com/plone/buildout.coredev/pull/954
https://github.com/plone/Products.CMFPlone/pull/4019
```

So coredev branch in combination with package PR will not work then?

File: Products/CMFPlone/tests/robot/test_controlpanel_site.robot [hunk 1]
```diff
@@ -94,8 +94,13 @@ then I can see a sitemap
 
     Import library    OperatingSystem
 
-    Download    ${PLONE_URL}/sitemap.xml.gz    saveAs=/tmp/sitemap.xml.gz
-    File Should Exist    /tmp/sitemap.xml.gz
+    # this is for robotframework browser > 17
+    # Download    ${PLONE_URL}/sitemap.xml.gz    saveAs=/tmp/sitemap.xml.gz
```
Comment (petschki) [inline Products/CMFPlone/tests/robot/test_controlpanel_site.robot:98]: We should go with the version >=18 and add it as test requirement here. Due to the fact that the tests are flaky event after backporting the version I'd rather merge the updated version in coredev (see https://github.com/plone/buildout.coredev/pull/954) and continue getting this here done ... @mauritsvanrees what do you think?

**Review by petschki — CHANGES_REQUESTED** [review]

File: Products/CMFPlone/tests/robot/test_controlpanel_site.robot (hunk 1 above)
Comment (1letter) [reply Products/CMFPlone/tests/robot/test_controlpanel_site.robot]: I updated the test to the rfb 18 version

**Comment by petschki** [comment]
robotframework-browser 18.x version pins are merged in coredev#6.1 and nvm is now activated for all jobs on jenkins. lets see what the tests say.

**Review by petschki — APPROVED** [review]

**Review by mauritsvanrees — APPROVED** [review]
Wow, that is a lot of work.  Thanks!
I skimmed through the changes, they seem mostly simplifying tests, removing all kinds of 'wait until' because they are no longer needed, which is good.

Jenkins 3.12 passes.  I have started 3.10.  Once that is green, I think this is mergeable!

**Comment by mauritsvanrees** [comment]
Or do one or more of the Jenkins nodes still need an update?

BTW, node 3 is down currently, "no route to host".

**Comment by 1letter** [comment]
@mauritsvanrees sometimes one or two tests fails on jenkins, locally all is fine. it looks like scarce resources on the nodes. but in general, the tests are stable.

**Comment by petschki** [comment]
@mauritsvanrees I had a discord session yesterday night with @gforcada where he fixed the jenkins configs to correctly install `nvm` for every job so this is done. Node 3 unfortunately has resource problems so it is offline regularly ... Node 4 also startet to cancel jobs because of resource problems.

I'm not sure if the resource problems are related to the new `rfbrowser` setup?

**Comment by 1letter** [comment]
@mauritsvanrees 
i started the 3.10 again with more parameters
```
https://github.com/plone/Products.CMFPlone/pull/4019
https://github.com/plone/plone.app.contenttypes/pull/704
https://github.com/plone/plone.app.discussion/pull/214
https://github.com/plone/plone.app.multilingual/pull/460
https://github.com/plone/plone.schemaeditor/pull/118
https://github.com/plone/plone.app.dexterity/pull/394
https://github.com/plone/plone.app.event/pull/409
```

**Comment by petschki** [comment]
I've merged all test-porting PRs and will fix the one failing test here. So no more extra parameters needed!

**Comment by petschki** [comment]
@jenkins-plone-org please run jobs

**Pull request merged**